go/plugins/localvec/localvec.go

```diff
@@ -134,7 +134,7 @@ func (r *retriever) Index(ctx context.Context, req *ai.IndexerRequest) error {
 	}
 
 	// Update the file every time we add documents.
-	tmpname := r.filename + ".tmp"
+	tmpname := r.filename
```

Comment:
I don't see how this "makes locavec use the same file across runs" more than it did before. Either way it's a fixed filename, right?

Reply:
Well, index updates are made in a different file and do not appear in the retrieved docs if you recreate the retriever. It is a filename fix though. I was describing the effect of the bug.

Reply:
Argh, the code is missing a rename of the .tmp file to the original file.  It should write to the temporary file and then call `os.Rename` to rename that to the real file.  Otherwise we are in trouble if the program crashes while writing to the temporary file.  Sorry about that.  I can send a fix tomorrow if nobody does it before then.

Reply:
I can do that. Thanks!